includes/class-wc-payments.php

```diff
@@ -789,16 +800,29 @@ public static function maybe_register_platform_checkout_hooks() {
 	public static function ajax_init_platform_checkout() {
 		$session_cookie_name = apply_filters( 'woocommerce_cookie', 'wp_woocommerce_session_' . COOKIEHASH );
 
+		$user        = wp_get_current_user();
+		$customer_id = self::$customer_service->get_customer_id_by_user_id( $user->ID );
+		if ( null === $customer_id ) {
+			// create customer.
+			$customer_data = WC_Payments_Customer_Service::map_customer_data( null, new WC_Customer( $user->ID ) );
+			self::$customer_service->create_customer_for_user( $user, $customer_data );
+		}
```

Comment:
This approach of creating the Customer like this before seems like the simplest way to get what we want:

- A customer ID we can pass to the platform checkout.
- A customer ID associated with the customer's current session — even if they're checking out as a guest.

Some alternative approaches that might be worth looking into;

- A re-usable function we can use throughout WCPay to initialize a customer.
- Creating the customer on the platform checkout side and just pass `null` here if the customer ID hasn't been created yet.

The current implementation work, so these are mostly notes that would be best to resolve and discuss in a separate issue.